pkgs/development/ocaml-modules/nonstd/default.nix

```diff
@@ -11,8 +11,12 @@ buildDunePackage rec {
     sha256 = "0ccjwcriwm8fv29ij1cnbc9win054kb6pfga3ygzdbjpjb778j46";
   };
 
-  useDune2 = lib.versionAtLeast ocaml.version "4.12";
-  postPatch = lib.optionalString useDune2 "dune upgrade";
+  duneVersion = if lib.versionAtLeast ocaml.version "4.12" then "3" else "1";
+  # patch to build with dune instead of jbuild
+  patches = fetchpatch {
```

Comment:
Can we include the patch locally in nonstd/?

Reply:
> Can we include the patch locally in nonstd/?

It is preferred at nixpkgs to use fetchpatch instead of including patches directly in nixpkgs. Same with lock files.

Reply:
I see thanks. My worry was with a download link becoming unavailable, but then I was kindly reminded that fetchpath would be available in cache.nixos.org.